src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionFixAllResolveHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions
+{
+    [ExportCSharpVisualBasicStatelessLspService(typeof(CodeActionFixAllResolveHandler)), Shared]
+    [Method("codeAction/resolveFixAll")]
+    internal class CodeActionFixAllResolveHandler : ILspServiceDocumentRequestHandler<RoslynFixAllCodeAction, RoslynFixAllCodeAction>
+    {
+        private readonly ICodeFixService _codeFixService;
+        private readonly ICodeRefactoringService _codeRefactoringService;
+        private readonly IGlobalOptionService _globalOptions;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CodeActionFixAllResolveHandler(
+            ICodeFixService codeFixService,
+            ICodeRefactoringService codeRefactoringService,
+            IGlobalOptionService globalOptions)
+        {
+            _codeFixService = codeFixService;
+            _codeRefactoringService = codeRefactoringService;
+            _globalOptions = globalOptions;
+        }
+
+        public bool MutatesSolutionState => false;
+
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(RoslynFixAllCodeAction request)
+            => ((JToken)request.Data!).ToObject<CodeActionResolveData>()!.TextDocument;
+
+        public async Task<RoslynFixAllCodeAction> HandleRequestAsync(RoslynFixAllCodeAction request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.GetRequiredDocument();
+            var data = ((JToken)request.Data!).ToObject<CodeActionResolveData>();
+            Assumes.Present(data);
+
+            var options = _globalOptions.GetCodeActionOptionsProvider();
+
+            var codeActions = await CodeActionHelpers.GetCodeActionsAsync(
```

Comment:
does this compute the actions, or just get the last computed ones?

Reply:
It recomputes

Reply:
1. if it recomputes, we need to be resilient to the results being different, and must not assume we get the right data back.
2. instead of recomputing, why not just cache these results from teh last call?  That way we have the actual real instances to use, instead of trying to recompute them?

Reply:
after having spoken to David about this, we think that recomputing is safer since the actions will be up to date with the document. and given that code actions are working today and recompute it would be risky to change that, but we can look into it for a follow up PR. @dibarbet

Reply:
ok.  if it recomputes, then just write your code to be resilient to the results being different.